### PR TITLE
[DPE-4911]: Fix relations flaky tests

### PR DESCRIPTION
**Pull request by Gu1nness**
## Issue

The `new_relations` test fails sometimes because it fails to get the secret.
This happens when one of the units has not received the  RelationJoined event.
It ends up in [this state](https://github.com/canonical/mongodb-operator/actions/runs/10079182721/job/27871431796?pr=439#step:24:13)

## Solution

Ensuring that the message is good allows to ensure that the secret is set without modifying the test application.
This is done like it's done already in another test.

### CODE REVIEW

**Comment by Gu1nness** [comment]
Will change charm_internal_revision in the end because too many PR's waiting to get merged as well.

File: tests/integration/helpers.py [hunk 1]
```diff
@@ -439,3 +439,18 @@ async def wait_for_mongodb_units_blocked(
     for attempt in Retrying(stop=stop_after_delay(timeout), wait=wait_fixed(1), reraise=True):
         with attempt:
             await check_all_units_blocked_with_status(ops_test, db_app_name, status)
+
+
+def is_relation_joined(ops_test: OpsTest, endpoint_one: str, endpoint_two: str) -> bool:
```
Comment (MiaAltieri) [inline tests/integration/helpers.py:444]: thank you very much for centralising this

File: tests/integration/relation_tests/new_relations/test_charm_relations.py [hunk 2]
```diff
@@ -99,6 +99,16 @@ async def test_database_relation_with_charm_libraries(ops_test: OpsTest):
     )
     await ops_test.model.wait_for_idle(apps=APP_NAMES, status="active")
 
+    await ops_test.model.block_until(
+        lambda: is_relation_joined(
+            ops_test,
+            FIRST_DATABASE_RELATION_NAME,
+            "database",
```
Comment (MiaAltieri) [inline tests/integration/relation_tests/new_relations/test_charm_relations.py]: nit: can you please add a constant for "database" 🙏

**Review by MiaAltieri — DISMISSED** [review]

File: tests/integration/relation_tests/new_relations/test_charm_relations.py (hunk 2 above)
Comment (Gu1nness) [reply tests/integration/relation_tests/new_relations/test_charm_relations.py]: Done :)

**Review by MiaAltieri — DISMISSED** [review]

**Review by Mehdi-Bendriss — DISMISSED** [review]

**Review by MiaAltieri — APPROVED** [review]

**Review by Mehdi-Bendriss — APPROVED** [review]

**Pull request merged**